tests/test_sns.py

```diff
@@ -0,0 +1,120 @@
+                "Statement": [
+                    {
+                        "Sid": "SpecificAllow",
+                        "Effect": "Allow",
+                        "Principal": {
+                            "AWS": "arn:aws:iam::123456789012:root"
+                        },
+                        "Action": [
+                            "SNS:Subscribe"
+                        ],
+                        "Resource": topic_arn
+                    },
+                    {
+                        "Sid": "Public",
+                        "Effect": "Allow",
+                        "Principal": "*",
+                        "Action": [
+                            "SNS:GetTopicAttributes"
+                        ],
+                        "Resource": topic_arn
+                    }
+                ]
+            })
+        )
+
+        p = self.load_policy({
+            'name': 'sns-rm-matched',
+            'resource': 'sns',
+            'filters': [
+                {'TopicArn': topic_arn},
+                {'type': 'cross-account',
+                 'whitelist': ["123456789012"]}
+            ],
+            'actions': [
+                {'type': 'remove-statements',
+                 'statement_ids': 'matched'}]
+            },
+            session_factory=session_factory)
+        resources = p.run()
+
+        self.assertEqual([r['TopicArn'] for r in resources], [topic_arn])
+        data = json.loads(client.get_topic_attributes(TopicArn=resources[0]['TopicArn'])['Attributes']['Policy'])
+        
+        self.assertEqual(
+            [s['Sid'] for s in data.get('Statement', ())],
+            ['SpecificAllow'])
+
+    @functional
+    def test_sns_remove_named(self):
+        session_factory = self.replay_flight_data('test_sns_remove_named')
+        client = session_factory().client('sns')
+        name = 'test-sns-remove-named'
+        topic_arn = client.create_topic(Name=name)['TopicArn']
+        self.addCleanup(client.delete_topic, TopicArn=topic_arn)
+
+        client.set_topic_attributes(
+            TopicArn=topic_arn,
+            AttributeName="Policy",
+            AttributeValue=json.dumps({
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Sid": "WhatIsIt",
+                        "Effect": "Allow",
+                        "Principal": "*",
+                        "Action": ["SNS:GetTopicAttributes"],
+                        "Resource": topic_arn
+                    }
+                ]
+            })
+        )
+
+        p = self.load_policy({
+            'name': 'sns-rm-named',
+            'resource': 'sns',
+            'filters': [{'TopicArn': topic_arn}],
+            'actions': [
+                {'type': 'remove-statements',
+                 'statement_ids': ['WhatIsIt']}]
+            },
+            session_factory=session_factory)
+
+        resources = p.run()
+        self.assertEqual(len(resources), 1) 
```

Comment:
this test didn't actually verify anything wrt to the action being implemented...